core_backend/tests/api/test_import_content.py

```diff
@@ -0,0 +1,265 @@
+from io import BytesIO
+from typing import Generator
+
+import pandas as pd
+import pytest
+from fastapi.testclient import TestClient
+
+
+# to be used for the DB duplicate checks later
+@pytest.fixture(scope="function")
+def existing_content_id(
+    client: TestClient,
+    fullaccess_token: str,
+) -> Generator[str, None, None]:
+    response = client.post(
+        "/content",
+        headers={"Authorization": f"Bearer {fullaccess_token}"},
+        json={
+            "content_title": "Title in DB",
+            "content_text": "Text in DB",
+            "content_language": "ENGLISH",
+            "content_tags": [],
+            "content_metadata": {},
+        },
+    )
+    content_id = response.json()["content_id"]
+    yield content_id
+    client.delete(
+        f"/content/{content_id}",
+        headers={"Authorization": f"Bearer {fullaccess_token}"},
+    )
+
+
+def _dict_to_csv_bytes(data: dict) -> BytesIO:
+    """
+    Convert a dictionary to a CSV file in bytes
+    """
+
+    df = pd.DataFrame(data)
+    csv_bytes = BytesIO()
+    df.to_csv(csv_bytes, index=False)
+    csv_bytes.seek(0)
+
+    return csv_bytes
+
+
+@pytest.fixture
+def data_valid() -> BytesIO:
+    data = {
+        "content_title": ["csv title 1", "csv title 2"],
+        "content_text": ["csv text 1", "csv text 2"],
+    }
+    return _dict_to_csv_bytes(data)
+
+
+@pytest.fixture
+def data_empty_csv() -> BytesIO:
+    data: dict = {}
+    return _dict_to_csv_bytes(data)
+
+
+@pytest.fixture
+def data_no_rows() -> BytesIO:
+    data: dict = {
+        "content_title": [],
+        "content_text": [],
+    }
+    return _dict_to_csv_bytes(data)
+
+
+@pytest.fixture
+def data_missing_columns() -> BytesIO:
+    data = {
+        "wrong_column_1": ["Value 1", "Value 2"],
```

Comment:
You should test titles being only trailing spaces.

Reply:
done

Reply:
This was clever, caught a bug with this test!